### PR TITLE
add ignore build props, incorporate BattlefieldCSMessage directly

### PR DESCRIPTION
**Pull request by ajkroeg**

### CODE REVIEW

**Comment by ajkroeg** [comment]
setting `"ShowSalvageFloaties": true` in settings will show floaties; defaults to off

File: .gitignore [hunk 1]
```diff
@@ -264,3 +264,6 @@ paket-files/
 __pycache__/
 *.pyc
 /readme_images
+
+# Ignore the VS Directory.build.props file
+Directory.Build.props
```
Comment (CptMoore) [inline .gitignore:269]: there is a section starting at line 8 with user specific files including VS stuff, move there?!

File: source/Patches/AttackStackSequence_OnAttackComplete.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    {
+        public static bool Prepare() => Control.Instance.Settings.ShowSalvageFloaties;
+
+        [HarmonyPriority(Priority.Last)]
+        public static void Prefix(AttackStackSequence __instance, MessageCenterMessage message)
+        {
+            List<AttackDirector.AttackSequence> directorSequences = __instance.directorSequences;
+            if (directorSequences != null)
+            {
+                try
```
Comment (CptMoore) [inline source/Patches/AttackStackSequence_OnAttackComplete.cs:23]: remove the try / catch and add `[HarmonyWrapSafe]` next to the priority annotation

File: source/Patches/AttackStackSequence_OnAttackComplete.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    [HarmonyPatch(typeof(AttackStackSequence), "OnAttackComplete")]
+    public static class AttackStackSequence_OnAttackComplete
+    {
+        public static bool Prepare() => Control.Instance.Settings.ShowSalvageFloaties;
+
+        [HarmonyPriority(Priority.Last)]
+        public static void Prefix(AttackStackSequence __instance, MessageCenterMessage message)
+        {
+            List<AttackDirector.AttackSequence> directorSequences = __instance.directorSequences;
+            if (directorSequences != null)
```
Comment (CptMoore) [inline source/Patches/AttackStackSequence_OnAttackComplete.cs:21]: use early return, meaning negative and return. avoids nesting. other languages call this semantically a guard and they also ask to keep it flat

File: source/Patches/AttackStackSequence_OnAttackComplete.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                int num;
+                if (mech == null)
+                {
+                    num = 0;
+                }
+                else
+                {
+                    MechDef mechDef = mech.ToMechDef();
+                    Log.Main.Info?.Log("BattlefieldCSMessage " + mech.DisplayName + " Part Check:");
+                    bool flag2 = mech.MechDef.IsLocationDestroyed(ChassisLocations.LeftArm);
```
Comment (CptMoore) [inline source/Patches/AttackStackSequence_OnAttackComplete.cs:87]: you can inline all if(flagX) checks

File: source/Patches/AttackStackSequence_OnAttackComplete.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            }
+
+            public static int SalvageParts(Mech mech)
+            {
+                int num;
+                if (mech == null)
+                {
+                    num = 0;
+                }
+                else
```
Comment (CptMoore) [inline source/Patches/AttackStackSequence_OnAttackComplete.cs:83]: return 0 directly if mech == null , avoids the else/converts its to a guard/early return, removes the need for num local variable declaration separate from assignment

File: source/Patches/AttackStackSequence_OnAttackComplete.cs [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+using Org.BouncyCastle.Math.Raw;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CustomSalvage
+{
+    [HarmonyPatch(typeof(AttackStackSequence), "OnAttackComplete")]
```
Comment (CptMoore) [inline source/Patches/AttackStackSequence_OnAttackComplete.cs:12]: use nameof(AttackStackSequence.OnAttackComplete) to allow ctrl-click (Rider) to open the dissassembler directly.. makes faster to check what is being patched. also allows static analysis on if there is a typo

**Review by CptMoore — APPROVED** [review]
Many code quality issues but no glaring bugs

**Pull request merged**